### ISSUE TITLE
Improve the usability of `WebClient`

### ISSUE THREAD

**Issue by trustin** [line/armeria#2390]
From an internal feed back:

> What would the ideal API for building an HTTP request look like? Do you have some imaginary ideal example?

Using method chains like Spring WebClient is easy for me to write http request. Annotation pattern like Retrofit is also easy for me.I prefer the Spring WebClient way to annotation pattern.

> What is the library/API you think Armeria team should benchmark?

Spring WebClient

> What other features would you like to see in Armeria?

Prepare a method which maps HttpResponse to Java Object, like `bodyToMono<T>`

**Comment by trustin** [comment]
Related: #2148

**Comment by trustin** [comment]
https://www.reddit.com/r/java/comments/gcctln/which_http_client_framework_do_people_personally/

**Comment by ikhoon** [comment]
https://www.reddit.com/r/java/comments/gy8vi2/which_openfeign_client_is_the_best/

**Comment by trustin** [comment]
https://stackoverflow.com/questions/63660975/armeria-http-client-how-to-add-query-string-parameters#comment112614741_63664409

It may be useful if we can build/execute an `HttpRequest` fluently, so that, for example, add a query parameter easily.

```java
// Not the final form - just an idea
HttpRequest req =
    HttpRequest.builder(HttpMethod.GET, "/foo")
               .queryParam("bar", "2")
               .cookie(...)
               .build();
webClient.execute(req);

// and/or:
webClient.prepare(HttpMethod.GET, "/foo")
         .queryParam("bar", "2")
         .cookie(...)
         .execute(); // or .execute(String.class) for automatic conversion
```

**Comment by tumile** [comment]
I'd love to take a stab. An idea on how the `HttpRequestBuilder` may look like
```java
HttpRequestBuilder.[get,post,put,..]  // shortcut for method(HttpMethod.*)
                  .method(HttpMethod)
                  .path(String)
                  .mediaType(MediaType)
                  .headers(HttpHeaders)
                  .content(CharSequence|String|byte[]|HttpData)
                  .pathParams(Map<String, String>)
                  .queryParams(QueryParams)
                  .cookies(Cookies)
                  .timeout(long)
```
There may also be `queryParam` or `cookie` (singular) to add a single value. Then we can delegate all `HttpRequest.of` to builder. I think we can even deprecate the more [complex ones](https://github.com/line/armeria/blob/9d4d5864d979bfb888131ce88aa64ec4c75965f5/core/src/main/java/com/linecorp/armeria/common/HttpRequest.java#L125) in favor of builder pattern.

I'm still thinking on how `webClient.prepare()...execute()` would work. Maybe having another class (`WebClientRequestBuilder`?) that extends `HttpRequestBuilder` with an `execute`?

**Comment by trustin** [comment]
Yeah, having a class like WebClientRequestBuilder would be nice. We could extract the common methods in HttpRequestBuilder into some common interface or abstract class. What do you think, @tumile ?

**Comment by ikhoon** [comment]
https://www.mocklab.io/blog/which-java-http-client-should-i-use-in-2020/
